lib/generator/js.rb

```diff
@@ -271,6 +273,18 @@ def js_for_data_criteria(force_sources=nil)
         HQMF2JS::Generator.render_template('data_criteria', {'all_criteria' => @doc.specific_occurrence_source_data_criteria(force_sources).concat(@doc.all_data_criteria), 'measure_period' => @doc.measure_period})
       end
       
+      def js_for_variables()
+        variables_js = ""
+        variables_js += "hqmfjs.VARIABLES = function(patient, initialSpecificContext) {\n" 
+        @doc.source_data_criteria.each do |criteria|
+          if criteria.type == :derived
```

Comment:
so it doesn't tell the difference between specific/non-specific occurrence variables. There was a reason for that in a first-pass implementation I did, but I don't think it's relevant now. I can add in a check for that.

Regarding if it's sufficient for if a criteria is a variable, Eric might be a good person to ask about that. It looked to me like variables were described by `:derived`. Otherwise they were directly a criteria type. None of the other types seemed like they made sense to be a variable to me, and I didn't see any examples of it. I can do a pass at a more comprehensive check as well.